### PR TITLE
Add config option to omit root objects from models

### PR DESCRIPTION
**Pull request by ghjm**
In https://github.com/99designs/gqlgen/pull/2858, model generation was changed to include structs for root level objects Query, Mutation and Subscription.  As described in https://github.com/99designs/gqlgen/issues/2874, this is a breaking change for people who were also using [gqlgenc](https://github.com/Yamashou/gqlgenc) to generate client code.

This pull request introduces a new configuration parameter, `omit_root_models`, which suppresses generation of structs for root level objects.

I have:
 - [X] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [X] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

**Comment by StevenACoffman** [comment]
Thanks!

File: plugin/modelgen/models_test.go [hunk 1]
```diff
@@ -298,6 +298,24 @@ func TestModelGeneration(t *testing.T) {
...
 
+func TestModelGenerationOmitRootModels(t *testing.T) {
+	cfg, err := config.LoadConfig("testdata/gqlgen_omit_root_models.yml")
+	require.NoError(t, err)
+	require.NoError(t, cfg.Init())
+	p := Plugin{
+		MutateHook: mutateHook,
+		FieldHook:  DefaultFieldMutateHook,
+	}
+	//cfg.OmitRootModels = true
```
Comment (StevenACoffman) [inline plugin/modelgen/models_test.go:309]: Can you please either remove this line or add a space to the comment?

**Review by StevenACoffman — APPROVED** [review]
Thanks for doing this!

File: plugin/modelgen/models_test.go (hunk 1 above)
Comment (ghjm) [reply plugin/modelgen/models_test.go]: Oops, this was test code that should have been removed.  Updated!

**Pull request merged**